### ISSUE TITLE
Add support for internationalization.

### ISSUE THREAD

**Issue by Ismael-VC** [JuliaLang/julialang.github.com#187]
I´ve started translating the website (just finished translating the font page to spanish) using transifex, providing a base for others to translate the Julia website in a collaborative way to their native languages:

https://www.transifex.com/projects/p/julia-web/language/es/

According to transifex the procedure would require inserting this code fragment before the `</head>` tag in each of the pages targeted for translation:

```javascript
<script type="text/javascript">
    window.liveSettings = {
        api_key: "ab25420d08984953bd7435fd47506f96",
        picker: "top-right",
        detectlang: true,
        dynamic: true,
        autocollect: true
    };
</script>
<script type="text/javascript" src="//cdn.transifex.com/live.js"></script>
```

The default configuration should add a language pop up menu in the bottom right corner of the page.

Here is how the draft looks like:

![draft](https://cloud.githubusercontent.com/assets/4594825/5566985/ac61c996-8efe-11e4-84f6-9953c428ce4e.JPG)

Link to the disscusion thread at julia-users group:

https://groups.google.com/forum/#!topic/julia-users/UCKss8S9LB8

**Comment by Ismael-VC** [comment]
I don't how could we approach the integration process (don't know JavaScript or web programming in general), but I'm studying this documentation:

http://docs.transifex.com/developer/live/webmasters

What do you think? Could we make some tests?

**Comment by Ismael-VC** [comment]
Project completion *widgets*, I just found about this I think it looks very neat:

https://www.transifex.com/projects/p/julia-web/widgets

@ivarne this would be a nice way to notify users about the community oriented approach to translations and also about their completeness.

Code:

```html
<a target="_blank" style="text-decoration:none; color:black; font-size:66%" href="https://www.transifex.com/projects/p/julia-web/resource/httpjulialangorg/" 
title="See more information on Transifex.com">Top translations: julia-web » httpjulialangorg</a><br/>
<img border="0" src="https://www.transifex.com/projects/p/julia-web/resource/httpjulialangorg/chart/image_png"/><br/><a target="_blank" href="https://www.transifex.com/"><img border="0" src="https://ds0k0en9abmn1.cloudfront.net/static/charts/images/tx-logo-micro.646b0065fce6.png"/></a>
```
Result:

<a target="_blank" style="text-decoration:none; color:black; font-size:66%" href="https://www.transifex.com/projects/p/julia-web/resource/httpjulialangorg/" 
title="See more information on Transifex.com">Top translations: julia-web » httpjulialangorg</a><br/>
<img border="0" src="https://www.transifex.com/projects/p/julia-web/resource/httpjulialangorg/chart/image_png"/><br/><a target="_blank" href="https://www.transifex.com/"><img border="0" src="https://ds0k0en9abmn1.cloudfront.net/static/charts/images/tx-logo-micro.646b0065fce6.png"/></a>

**Comment by Ismael-VC** [comment]
@ivarne this is how it looks when the page is updated:

![update](https://cloud.githubusercontent.com/assets/4594825/5792697/89a081de-9eeb-11e4-8f1e-fc8757c61cbd.JPG)

In this example only the header was updated, and you can see that the string is left untranslated while the rest of the strings are translated.

I get a notification about the change in order to add the new string (all this is automatic!)

![addstring](https://cloud.githubusercontent.com/assets/4594825/5792736/fbbcf256-9eec-11e4-8790-69fffab51b71.JPG)

Here is the translation editor in action:

![translating](https://cloud.githubusercontent.com/assets/4594825/5792739/14533082-9eed-11e4-813e-39e7595a442a.JPG)

An here is the new result!

![listo](https://cloud.githubusercontent.com/assets/4594825/5792740/271d0346-9eed-11e4-8a60-c017d50c2c9f.JPG)


cheers

**Comment by ivarne** [comment]
This looks impressive to me.

What does others think about this?

**Comment by i2000s** [comment]
I feel that these tools can only translate content into another language, but what if I want to replace some link to a localized service provider? Or, what if I want to add or delete some content so that it can be better fit into the local habit of using a website for promoting Julia? Some content from local collaborators should appear on the localized version at least. I haven't look into those tools deeply, but can they realize the utilities above? Is there any better way for internationalization? I would just translate the whole website in another repo since JuliaLang.org is now licensed under MIT license, and customize the content over the new repo.

**Comment by Ismael-VC** [comment]
@i2000s the main issue for me is to stay in sync with the continuous updates *automagically*.

I will look into the other issues you mention, for now I only want an adequate entry point for Spanish speakers (mirroring the English version), perhaps in the future a broader array of localized *official* (user updated) Julia web pages can be made (under the JuliaLang umbrella)?

What I've been doing now is show the translated Julia page at Transifex to people that don't speak English, and use several other methods to distribute information, chat, stay in touch, etc.

### Julia for Spanish users:

* [julia-users-es](https://groups.google.com/forum/#!forum/julia-users-es): Google mailing list.
* [julialang-es](https://gitter.im/Ismael-VC/julialang-es): Gitter chat room.
* [Julia - Lenguaje de Programación](https://www.facebook.com/groups/julialang.es/): Facebook group.
* [Lenguaje de Programación Julia - México, D.F.](http://www.meetup.com/julialang-mx/): Meetup group (This is the only one that is specific for Mexico City, because of the nature of meetings, but I plan on holding virtual meetups (via Google hangouts), so perhaps I should change the name or make another meetup group for virtual meetings only).

Suggestions are always welcome.

**Comment by Ismael-VC** [comment]
Pull request to add *julialang-es* Gitter chat room: https://github.com/JuliaLang/julialang.github.com/pull/218

**Comment by ninjin** [comment]
The only concern I have is keeping all the version in sync.  It is really tough to keep eyes on all of the non-english communities and if the page is not up-to-date the image of the project will suffer, possibly without most of us even noticing.  In short, given fewer eyeballs, slipping up becomes easier.

A small survey regarding other languages and if they have internationalisation on their homepage, do correct me if I missed it in a menu somewhere:

* [Python](https://www.python.org/), no.
* [Ruby](https://www.ruby-lang.org), yes.
* [Rust](http://www.rust-lang.org/), no.
* [Go](https://golang.org/), no.

**Comment by Ismael-VC** [comment]
I created Julia-i18n for this purpose, to ensure quality, manage teams, update and sync resources, collaborate, have better i18n than most other languages, gather more eyes, etc.:

* https://gitter.im/JuliaLangEs/julia-i18n

These are Transifex Live configurations:

![captura de pantalla 2015-07-16 a las 07 29 10](https://cloud.githubusercontent.com/assets/4594825/8722675/9ebbbb14-2b8c-11e5-939d-32ce20a22993.png)

![captura de pantalla 2015-07-16 a las 07 29 39](https://cloud.githubusercontent.com/assets/4594825/8722774/96565bd6-2b8d-11e5-8798-bf28e3797bfd.png)

![captura de pantalla 2015-07-16 a las 07 29 25](https://cloud.githubusercontent.com/assets/4594825/8722778/9ec0e58e-2b8d-11e5-8ca3-16750d5f5c29.png)

![captura de pantalla 2015-07-16 a las 07 29 10](https://cloud.githubusercontent.com/assets/4594825/8722780/a47cf6c0-2b8d-11e5-888b-65e0301b02d0.png)

I'll try to set up a staging environment today, We could ask for a minimum number of members per translation team, translations could be submitted to production only when the page is 100% translated. Stagnated versions could be removed if interest is lost and re added again if this changes.

My intent is precisely to have more **quality** not less.

Also the Julia web page is not that big or complicated it can be translated in one day and reviewed in another day, after that it's just a matter of translating the changes on an almost real time basis, which is super easy.

CC: @yuyichao

**Comment by Ismael-VC** [comment]
Closed by #252 🎉
